crates/sp1/ev-exec/program/src/main.rs

```diff
@@ -106,8 +106,13 @@ pub fn main() {
 
     let mut cursor = 0;
     for (proof, root) in inputs.proofs.iter().zip(roots) {
-        assert!(proof.is_of_presence(), "NamespaceProof must be of presence");
-
+        // we also need to verify absence proofs because we get a root for them
+        if proof.is_of_absence() {
```

Comment:
I introduced this issue in https://github.com/celestiaorg/celestia-zkevm-hl-testnet/pull/188

There is a chance that this case was being hit previously but we were calling `verify_complete_namespace()` with an empty set of leaves against the root which would have allowed it succeed, but unbeknownst to us realising it was an absence proof being verified. 

This actually seems to me like more or a problem with the contains()` method on the namespace ids: https://docs.rs/nmt-rs/latest/src/nmt_rs/namespaced_hash.rs.html#345-352

And a more fitting name would probably be `is_in_range()`